operator/src/test/java/org/bf2/operator/managers/StrimziManagerTest.java

```diff
@@ -58,10 +66,11 @@ public class StrimziManagerTest {
 
     @BeforeEach
     public void beforeEach() {
-        // before each test clean Kubernetes server (no Deployments from other runs)
-        this.server.before();
+        try (NamespacedKubernetesClient client = this.server.getClient().inAnyNamespace()) {
```

Comment:
Without this change the test was failing because the StrimziManagerTest instance and the StrimziManager instance had clients pointing at different mock kubernetes servers on different ports. The test created a lease in one mock server and the implementation tried to delete it from another which was a no-op.

Reply:
that sounds like we've got really fragile test code.  @shawkins are you aware of this problem?  Is there a nicer way?

Reply:
We don't want to call before.  Until we get to a later version of fabric8, you mostly need to clean the state of the mock server yourself in-between tests.  Due to a quarkus / okhttp issue we cannot scope the mock server to test classes - it is scoped to the whole module.  In fabric8 6.x there is a KuberentesMockServer reset method that should make things easy regardless.